summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormPlatform.scala

```diff
@@ -196,9 +216,31 @@ abstract class Storm(options: Map[String, Options], transformConfig: Summingbird
       StormStatProvider.registerMetrics(jobID, context, countersForSpout)
       SummingbirdRuntimeStats.addPlatformStatProvider(StormStatProvider)
     })
-    val stormSpout = tormentaSpout.openHook(registerAllMetrics.get).getSpout
-    val parallelism = getOrElse(stormDag, node, parOpt.getOrElse(DEFAULT_SOURCE_PARALLELISM)).parHint
-    topologyBuilder.setSpout(nodeName, stormSpout, parallelism)
+    val hookedTormentaSpout = tormentaSpout.openHook(registerAllMetrics.get)
+
+    val summerOpt: Option[SummerNode[Storm]] = stormDag.dependantsOf(node.asInstanceOf[StormNode]).collect { case s: SummerNode[Storm] => s }.headOption
```

Comment:
I don't see how this is guarding against the case we are testing, and I'm afraid the order of the dependants is saving you. It seems to me you need to check that *ALL* the dependants are `SummerNode[Storm]` here, not just that there is at least one. Can you explain how this is safe? I don't understand.